modules/gateways/blockonomics/blockonomics.php

```diff
@@ -281,6 +266,22 @@ public function convertFiatToBlockonomicsCurrency($fiat_amount, $currency, $bloc
         return intval(1.0e8 * $fiat_amount / $price);
     }
 
+    /**
+     * Convert received btc percentage to invoice currency
+     *
+     * @param array $order
+     * @param string $percentPaid
+     * @return float converted value
+     */
+    public function convertPercentPaidToInvoiceCurrency($order, $percentPaid)
+    {
+        if (floatval($order['basecurrencyamount']) == 0) {
+            $order['basecurrencyamount'] = $order['value'];
+        }
```

Comment:
Better compare >0 . Direct comparison is not reliable in float (may not apply in Decimal but still)
Don't change `$order['basecurrencyamount']`  as that would be tainting the data. We should use another variable that uses *basecurrency* amount if it is greater than 0 else uses *value*

Reply:
Restructured in ce46f25